twig/contracts/registry.v.py

```diff
@@ -0,0 +1,206 @@
+# todo
+# delete pkgs
+
+# Events
+Release: event({_package: indexed(bytes32), _version: bytes32, _uri: bytes32})
+
+owner: public(address)
+
+# Package Data: (package_id => value)
+packages: public(
+    {
+        exists: bool,
+        createdAt: timestamp,
+        updatedAt: timestamp,
+        name: bytes32,
+        releaseCount: int128,
+    }[bytes32]
+)
+
+#  Release Data: (release_id => value)
+releases: public(
+    {
+        exists: bool,
+        createdAt: timestamp,
+        packageId: bytes32,
+        version: bytes32,
+        uri: bytes32,
+    }[bytes32]
+)
+
+
+# package_id#release_count => release_id
+packageReleaseIndex: bytes32[bytes32]
+# Total number of packages in registry
+packageCount: public(int128)
+# Total number of releases in registry
+releaseCount: public(int128)
+# Total package number (int128) => package_id (bytes32)
+packageIds: bytes32[int128]
+# Total release number (int128) => release_id (bytes32)
+releaseIds: bytes32[int128]
+
+EMPTY_BYTES: bytes32
+
+
+@public
+def __init__():
+    self.owner = msg.sender
+
+
+@public
+def transferOwner(newOwner: address):
+    """
+    Change ownership of contract.
+    """
+    assert self.owner == msg.sender
+    self.owner = newOwner
+
+
+@public
+def getReleaseId(packageName: bytes32, version: bytes32) -> bytes32:
+    """
+    Return the `release_id` associated with a given package name and release version.
+    """
+    releaseConcat: bytes[64] = concat(packageName, version)
+    releaseId: bytes32 = sha3(releaseConcat)
+    assert self.releases[releaseId].exists
+    return releaseId
+
+
+@public
+def generateReleaseId(packageName: bytes32, version: bytes32) -> bytes32:
+    releaseConcat: bytes[64] = concat(packageName, version)
+    releaseId: bytes32 = sha3(releaseConcat)
+    return releaseId
+
+
+@public
+def getPackageName(packageId: bytes32) -> bytes32:
+    assert self.packages[packageId].exists
+    return self.packages[packageId].name
+
+
+@public
+def getAllPackageIds(
+    offset: int128, length: int128
+) -> (bytes32, bytes32, bytes32, bytes32, bytes32):
+    # todo update to uint256?
+    assert length == 5
```

Comment:
I assume this is how you make this compliant.  I'd be curious if there's a way to loosen the restriction such that `length <= 5`.  Doesn't have to block you moving this forward.

Reply:
You can return fixed arrays I think. You can definitely return `bytes[160]`, if not `bytes32[5]`